guides/micronaut-flyway/micronaut-flyway.adoc

```diff
@@ -0,0 +1,193 @@
+
+Configure the database migrations directory for http://www.flyway.org[Flyway] in `application.yml`.
+
+resource:application.yml[tag=flyway]
+
+<1> Enable flyway for the `default` datasource.
+
+NOTE: Micronaut flyway integration works with multiple datasources.
+
+Create the following files with the database schema creation:
+
+resource:db/migration/V1__create-person.sql[]
+
+During application startup, http://www.flyway.org[Flyway] executes the SQL file and creates the schema needed for the application.
+
+If you check the database schema, there are two tables:
+
+- `person`
+- `flyway_scheme_history`
+
+Flyway uses the table `flyway_scheme_history` to keep track of database migrations.
+
+The `person` table looks like:
+
+|===
+|Column | Nullable
+
+|`id`
+|NO
+
+|`version`
+|NO
+
+|`name`
+|NO
+
+|`age`
+|NO
+|===
+
+=== Drop Not Null Constraint
+
+Applications change. Make `age` optional:
+
+source:Person[tags=nullableage]
+
+Add a new migration to drop the null constraint:
+
+resource:db/migration/V2__nullable-age.sql[]
+
+After the migration, the `person` table looks like:
+
+|===
+|Column | Nullable
+
+|`id`
+|NO
+
+|`version`
+|NO
+
+|`name`
+|NO
+
+|`age`
+|YES
+|===
+
+== Flyway endpoint
+
+To enable the Flyway endpoint, add the `management` dependency on your classpath.
+
+dependency:micronaut-management[]
+
+Enable the Flyway endpoint:
+
+resource:application.yml[tag=endpoints]
+
+include::{commondir}/common-testcontainers-postgres.adoc[]
+
+=== Test
+
+Create a test that invokes the Flyway endpoint
+
+test:FlywayEndpointTest[]
+
+callout:micronaut-test[1]
+callout:http-client[2]
+
+=== Running the application
+
+include::{commondir}/common-run-postgres-with-docker.adoc[]
+
+Set up the following environment variables to connect to the PostgreSQL database you started with Docker.
+
+resource:application.yml[tag=datasource]
+
+<1> The JDBC URL matches the database name you used in the previous command.
+<2> Use PostgreSQL driver.
+<3> Configure the PostgreSQL dialect.
+<3> You handle database migrations via Flyway
+
+[source,bash]
+----
+export DATASOURCES_DEFAULT_USERNAME=dbuser
+export DATASOURCES_DEFAULT_PASSWORD=theSecretPassword
+export VAT_PERCENTAGE=20
+----
+
+Configure your default datasource to use the PostgreSQL database you started with Docker:
+
+include::{commondir}/common-runapp-instructions.adoc[]
+
+You can run a cURL command to test the application:
+
+[source, bash]
+----
+$ curl http://localhost:8080/flyway
+----
+
+You will see information about migrations.
+
+include::{commondir}/common-graal-with-plugins.adoc[]
+
+You can run a cURL command to test the application:
+
+[source, bash]
+----
+$ curl http://localhost:8080/flyway
+----
+
+You will see information about migrations.
+
+include::{commondir}/common-next.adoc[]
+
+Check https://micronaut-projects.github.io/micronaut-flyway/latest/guide/[Micronaut Flyway] integration.
+
+Learn more about (https://flywaydb.org)[Flyway].
```

Comment:
Remove the parenthesis. Not needed in Asciidoctor and the output is rendered as:
![image](https://user-images.githubusercontent.com/559192/142833120-1b4ff5a6-8009-4d90-b16d-cb1ae596265b.png)